packages/host/src/lib/cpm-connector.ts

```diff
@@ -358,21 +342,25 @@ export class CPMConnector extends TypedEmitter<Events> {
         connection.socket
             .on("close", async () => { await this.handleConnectionClose(); });
 
+        /**
+         * @TODO: Distinguish existing `connect` request and started communication (Manager handled this host
+         * and made requests to it).
+         * @TODO: Provide detailed communication status.
+        */
+
         this.connected = true;
         this.connectionAttempts = 0;
 
-        this.registerChannels();
-
-        connection.req.on("error", (error: any) => {
+        connection.req.on("error", async (error: any) => {
             this.logger.error("Request error", error);
 
-            this.reconnect();
+            await this.reconnect();
```

Comment:
```suggestion
            await this.reconnect();
```

Try catch is needed here, otherwise STH may crash. The same with all the other aspects.

If STH should crash in that case, a crash handler should be called in main class.